src/foundry/foundry.js/pixi/containers/placeableObject.d.ts

```diff
@@ -1,425 +1,418 @@
+     */
+    clone(): this;
 
-  /**
-   * Clone the placeable object, returning a new object with identical attributes
-   * The returned object is non-interactive, and has no assigned ID
-   * If you plan to use it permanently you should call the create method
-   *
-   * @returns A new object with identical data
-   */
-  clone(): PlaceableObject<D>;
+    /** @override */
+    destroy(options?: Parameters<PIXI.Container['destroy']>[0]): void;
 
-  /**
-   * Draw the placeable object into its parent container
-   * @returns The drawn object
-   */
-  abstract draw(): Promise<this>;
+    /**
+     * Draw the placeable object into its parent container
+     * @returns The drawn object
+     */
+    abstract draw(): Promise<this>;
 
-  /**
-   * Draw the primary Sprite for the PlaceableObject
-   */
-  protected _drawPrimarySprite(texture: PIXI.Texture): PIXI.Sprite | null;
+    /**
+     * Refresh the current display state of the Placeable Object
+     * @returns The refreshed object
+     */
+    abstract refresh(): this | void;
 
-  /**
-   * Refresh the current display state of the Placeable Object
-   * @returns The refreshed object
-   */
-  abstract refresh(): unknown;
+    /**
+     * Register pending canvas operations which should occur after a new PlaceableObject of this type is created
+     */
+    protected _onCreate(): void;
 
-  static create<T extends PlaceableObject, U>(
-    this: ConstructorOf<T>,
-    data: Expanded<U> extends DeepPartial<T['data']> ? U : DeepPartial<T['data']>,
-    options?: Entity.CreateOptions
-  ): Promise<T | void>;
-  static create<T extends PlaceableObject, U>(
-    this: ConstructorOf<T>,
-    data: Expanded<U> extends DeepPartial<T['data']> ? U[] : DeepPartial<T['data']>[],
-    options?: Entity.CreateOptions
-  ): Promise<T | T[] | void>;
+    /**
+     * Define additional steps taken when an existing placeable object of this type is updated with new data
+     * @param options - (unused)
+     * @param userId  - (unused)
+     */
+    protected _onUpdate(changed: DeepPartial<D['data']>, options?: any, userId?: string): void;
 
-  update<U>(data: Expanded<U> extends DeepPartial<D> ? U : never, options?: Entity.UpdateOptions): Promise<this>;
-  update(data: DeepPartial<D>, options?: Entity.UpdateOptions): Promise<this>;
+    /**
+     * Define additional steps taken when an existing placeable object of this type is deleted
+     * @param options - (unused)
+     * @param userId  - (unused)
+     */
+    protected _onDelete(options: any, userId: string): void;
 
-  delete(options?: Entity.DeleteOptions): Promise<this>;
+    /**
+     * Assume control over a PlaceableObject, flagging it as controlled and enabling downstream behaviors
+     * @param options - Additional options which modify the control request
+     *                  (default: `{}`)
+     * @returns A flag denoting whether or not control was successful
+     */
+    control(options?: PlaceableObject.ControlOptions): boolean;
```

Comment:
consider getting rid of the namespace and moving the `ControlOptions` outside of the `declare global`.

Reply:
I considered that. And I know that this interface does not exist in foundry.
But personally, I would prefer to let it exist.
Because from a user perspective, how would you override this method?
`control(options?: Paremeters<PlaceableObject['control']>[0]): boolean`
Looks very unsexy …